server/service/src/stock_line/update.rs

```diff
@@ -168,6 +193,45 @@ fn generate_location_movement(
     Ok(movement)
 }
 
+fn generate_barcode_row(
+    connection: &StorageConnection,
+    existing: StockLineRow,
+    value: String,
+) -> Option<BarcodeRow> {
+    // for an empty string, simply unlink the barcode
+    if value.is_empty() {
+        return None;
+    }
+
+    let filter = BarcodeFilter::new()
+        .item_id(EqualFilter::equal_to(&existing.item_id))
+        .pack_size(EqualFilter::equal_to_i32(existing.pack_size));
+
+    let existing_rows = BarcodeRepository::new(connection).query_by_filter(filter);
+
+    if existing_rows.is_err() {
```

Comment:
We usually treat errors in database as internal error (rather then silently ignore)

Reply:
yeah, I thought you wouldn't like this. My thinking was that I didn't want a barcode update to cause the whole stock line update to fail. I've changed my mind, will now change the code too.